temporal-test-server/src/main/java/io/temporal/internal/testservice/TestWorkflowMutableStateImpl.java

```diff
@@ -3143,6 +3159,30 @@ private static PendingNexusOperationState convertNexusOperationState(
     }
   }
 
+  private static NexusOperationCancellationState convertNexusOperationCancellationState(
+      State state, NexusOperationData data) {
+    // Terminal states have already been filtered out, so only handle pending states.
+    if (data.getAttempt() > 1) {
```

Comment:
If attempt is greater than 1 that means we are retrying which I thought meant we always return backing off

Reply:
No backing off is for between attempts. If a nexus task is currently scheduled, we should report that correctly.

Reply:
Ah okay, I've updated to track that. It's hard to write functional tests for that logic though because the test server allows time skipping while waiting for retries.

Reply:
WDYM updated to track that? I think it's fine if the tests gloss over this detail to some degree. E.g. you can accept either scheduled or backing off.

Reply:
I added a new field to track whether the operation is waiting on retry. So it should be accurate now.